lazyscribe/artifacts/sklearn.py

```diff
@@ -0,0 +1,118 @@
+"""Joblib-based handler for scikit-learn objects."""
```

Comment:
I was tinkering with this in testing and this handler actually worked for non-sklearn objects too. Any reason why it's recommended/named only for sklearn objects? I was going to suggest a generic handler for pickling nearly any python object but this seems to do something similar (not super familiar with joblib tbh).

Reply:
I imagine we could but we risk falling prey to pickling and unpickling with different package versions of the pickled object.
Maybe for a `GenericArtifact` class, we could get the object's package name (in addition to the package version) and store it as an artifact attribute which can be used for version matching when loading.

Reply:
Interesting. I do like the idea of focusing on the method of serialization and not the serialized object itself. Doing a quick search, we could do something like

1. Get the package the value belongs to with `value.__module__`, and
2. Use `importlib.metadata.version` or `pip freeze | grep <module_name>` to find the package version.

I wonder how that solution handles versioning for packages that have a different name for the import path and the distribution (e.g. `scikit-learn`).

Reply:
Quick follow up -- you can use [`importlib_metadata.packages_distributions`](https://docs.python.org/3/library/importlib.metadata.html#mapping-import-to-distribution-packages) to map the import path to the distribution name. Using the distribution name, `importlib_metadata.version` can provide the package version.